client/app/dbaas/logs/detail/options/logs-options.service.js

```diff
@@ -131,6 +126,16 @@ class LogsOptionsService {
         return _.filter(options, option => option.quantity > 0);
     }
 
+    _filterOptions (options, max) {
+        return options
+            .filter(option => option[max] > 0)
+            .map(option => {
```

Comment:
.map could be replaced by .forEach in this case.

Reply:
I understand why you are asking me to use .forEach instead of .map...... in general we think forEach is fast because it returns the same array. But in most cases map is fastest. Please check this https://codeburst.io/javascript-map-vs-foreach-f38111822c0f

Reply:
Maintainability > Performance.
That said, you are changing the option, so you're right, it should be a .map